lib/utils.js

```diff
@@ -148,84 +154,96 @@ export function getDefaultAccount() {
+      this.contract = contract;
+      for (const i in this.contract) {
+        if (this[i] === undefined) {
+          this[i] = this.contract[i];
+        }
       }
+      // for (var prop in this.contract) {
+      //   if (!this.hasOwnProperty(prop)) {
+      //     this[prop] = superclass[prop];
+      //   }
+      // }
     }
-    // for (var prop in this.contract) {
-    //   if (!this.hasOwnProperty(prop)) {
-    //     this[prop] = superclass[prop];
-    //   }
-    // }
-  }
 
-  static async new() {
-    return superclass.new()
-      .then((contract) => {
-        return new this(contract);
-      }, (ex) => { throw ex; });
-  }
+    static async new() {
+      return superclass.new().then(
+        contract => {
+          return new this(contract);
+        },
+        ex => {
+          throw ex;
+        }
+      );
+    }
 
-  static async at(address) {
-    return superclass.at(address)
-      .then((contract) => {
-        return new this(contract);
-      }, (ex) => { throw ex; });
-  }
+    static async at(address) {
+      return superclass.at(address).then(
+        contract => {
+          return new this(contract);
+        },
+        ex => {
+          throw ex;
+        }
+      );
+    }
 
-  static async deployed() {
-    return superclass.deployed()
-      .then((contract) => {
-        return new this(contract);
-      }, (ex) => { throw ex; });
-  }
+    static async deployed() {
+      return superclass.deployed().then(
+        contract => {
+          return new this(contract);
+        },
+        ex => {
+          throw ex;
+        }
+      );
+    }
 
-  /**
-   * Call setParameters on this scheme.
-   * Returns promise of parameters hash.
-   * If there are any parameters, then this function must be overridden by the subclass to provide them.
-   * @param overrides -- object with properties whose names are expected by the scheme to correspond to parameters.  Overrides the defaults.
-   *
-   * Should have the following properties:
-   *
-   *  for all:
-   *    voteParametersHash
-   *    votingMachine -- address
-   *
-   *  for ContributionReward:
-   *    orgNativeTokenFee -- number
-   *    schemeNativeTokenFee -- number
-   */
-  // eslint-disable-next-line no-unused-vars
-  async setParams(params) {
-    return await '';
-  }
+    /**
+     * Call setParameters on this scheme.
+     * Returns promise of parameters hash.
+     * If there are any parameters, then this function must be overridden by the subclass to provide them.
+     * @param overrides -- object with properties whose names are expected by the scheme to correspond to parameters.  Overrides the defaults.
+     *
+     * Should have the following properties:
+     *
+     *  for all:
+     *    voteParametersHash
+     *    votingMachine -- address
+     *
+     *  for ContributionReward:
+     *    orgNativeTokenFee -- number
+     *    schemeNativeTokenFee -- number
+     */
+    // eslint-disable-next-line no-unused-vars
```

Comment:
Why disable eslint here ?

Reply:
because there is an unused var, but that is intentional

Reply:
how come unused var is intentional ?

Reply:
it is like an abstract method, never directly used

Reply:
so how about  : 
async setParams(params) {
    return await params;
  }
instead of : 
async setParams(params) {
    return await "";
  }
If its any way never directly used.

Reply:
I guess we should try to avoid linter exceptions if we can

Reply:
That suggestion would return the wrong datatype and thus be misleading to anyone trying to understand the code.

So I did it like this:

```
     */
    async setParams(params) {
      params; // avoid lint error
      return await "";
    }
```

Reply:
ok